enterprise/backend/src/metabase_enterprise/caching/api.clj

```diff
@@ -0,0 +1,118 @@
+(ns metabase-enterprise.caching.api
+  (:require
+   [compojure.core :refer [GET]]
+   [metabase.api.common :as api]
+   [metabase.api.common.validation :as validation]
+   [metabase.api.routes.common :refer [+auth]]
+   [metabase.events :as events]
+   [metabase.util :as u]
+   [metabase.util.cron :as u.cron]
+   [metabase.util.i18n :refer [tru]]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(defn- assert-valid-models [model ids]
+  (if (= model "root")
+    (when-not (zero? (first ids))
+      (throw (ex-info (tru "Root configuration is only valid with model_id = 0") {:status-code 400
+                                                                                  :model_id    (first ids)})))
+    (api/check-404 (t2/select-one (case model
+                                    "database"  :model/Database
+                                    "dashboard" :model/Dashboard
+                                    "question"  :model/Card)
+                                  :id [:in ids]))))
+
+(defn- audit-caching-change! [id target prev new]
+  (events/publish-event!
+   :event/caching-update
+   {:user-id api/*current-user-id*
+    :model   :model/CacheConfig
+    :details {:id             id
+              :model          (:model target)
+              :model_id       (:model_id target)
+              :previous-value prev
+              :next-value     new}}))
+
+(api/defendpoint GET "/"
+  "Return cache configuration."
+  [:as {{:strs [model collection]
+         :or   {model "root"}}
+        :query-params}]
+  {model      (ms/QueryVectorOf [:enum "root" "database" "dashboard" "question"])
+   collection [:maybe ms/PositiveInt]}
+  (validation/check-has-application-permission :setting)
+  (let [items (t2/select :model/CacheConfig
+                         :model [:in model]
```

Comment:
`model` is going to be a vector here, it's coerced by `QueryVectorOf`. So you can query for cards + dashboards with a same query here.

Reply:
Gotcha. 

Optional: As silly as it may seem I think it's worth rebinding to `models` in the first `let` block then with a comment explaining this. I can definitely seem myself tripping over things like this again, and can imagine the same is true for others.